File: arrays/longest_consecutive_sequence/longest_consecutive_sequence.ts
export class Solution {
    /**
    * @param {string} testInput
    * @return {any} Returns either boolean to validate the test cases or a custom result. 
    */
    run_solution(testInput: number[]) {
        const sorted:number[] = [...new Set(testInput.sort((a, b) => a - b))]
        let longest = 1;
        let currSequence = 1
        if(testInput.length === 0) {
            return 0
        }
        sorted.forEach((num,i) => {
            if(num-sorted[i-1] ===1) {
                currSequence++
            } else {
                currSequence = 1
            }
            longest = Math.max(longest, currSequence)
        })
    

        return longest
    }
}

File: arrays/longest_consecutive_sequence/longest_consecutive_sequence.py
""" Solution Module """


class Solution:
    """ Solution Class """

    def run_solution(self, test_input):
        """ Solutio method"""
        sorted_list = sorted(list(set(test_input)))
        print(sorted_list)
        if len(test_input) == 0:
            return 0

        longest = 1
        current = 1

        for i, num in enumerate(sorted_list):
            if num-sorted_list[i-1] == 1:
                current += 1
            else:
                current = 1

            longest = max(longest, current)

        return longest
